x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/optimizer/Optimizer.java

```diff
@@ -154,14 +174,168 @@ protected LogicalPlan rule(UnaryPlan plan) {
             return plan;
         }
     }
-    
+
+    static class SkipQueryOnLimitZero extends org.elasticsearch.xpack.ql.optimizer.OptimizerRules.SkipQueryOnLimitZero {
+
+        @Override
+        protected LogicalPlan skipPlan(Limit limit) {
+            return Optimizer.skipPlan(limit);
+        }
+    }
+
+    private static LogicalPlan skipPlan(UnaryPlan plan) {
+        return new LocalRelation(plan.source(), plan.output());
+    }
+
+    /**
+     * Combine tail and head into one limit.
+     * The rules moves up since the first limit is the one that defines whether it's the head (positive) or
+     * the tail (negative) limit of the data and the rest simply work in this space.
+     */
+    static final class CombineLimits extends OptimizerRule<LimitWithOffset> {
+
+        CombineLimits() {
+            super(TransformDirection.UP);
+        }
+
+        @Override
+        protected LogicalPlan rule(LimitWithOffset limit) {
+            if (limit.child() instanceof LimitWithOffset) {
```

Comment:
```
if (limit.child() instanceof LimitWithOffset == false) {
   return limit;
}
```

Reply:
I've modified it since I think the gains (less formatting) are sufficient. In general our rules tend to start with if() and I like consistency.
That and having only one `return`, essentially a clear exit vs multiple return paths (which is why I haven't updated `PushDownOrderBy`). This is mainly a styling issue but then again, we're software writers not just engineers.